### PR TITLE
#1208 Check availability of Istio installation

### PR DESCRIPTION
**Pull request by agrimmer**

### CODE REVIEW

File: installer/scripts/common/setupIstio.sh [hunk 1]
```diff
@@ -1,23 +1,39 @@
 #!/bin/bash
 source ./common/utils.sh
 
-kubectl create namespace istio-system
+kubectl get ns istio-system
+ISTIO_AVAILABLE=$?
+if [[ "$ISTIO_AVAILABLE" == 0 ]] && [[ "$ISTIO_INSTALL_OPTION" == "Reuse" ]]; then
+    # An istio-version is already installed
+    print_info "Istio installation is reused but its compatibility is not checked"
+elif [[ "$ISTIO_AVAILABLE" == 0 ]] && ([[ "$ISTIO_INSTALL_OPTION" == "StopIfInstalled" ]] || [[ "$ISTIO_INSTALL_OPTION" == "" ]]); then
```
Comment (christian-kreuzberger-dtx) [inline installer/scripts/common/setupIstio.sh]: Please also handle 
```bash
[ "$ISTIO_INSTALL_OPTION" == "ISTIO_INSTALL_OPTION_PLACEHOLDER" ]]
```
to remain compatibility with older CLIs

(not sure if it goes into this if statement or into the else branch)

File: installer/scripts/common/setupIstio.sh [hunk 2]
```diff
@@ -1,23 +1,39 @@
 #!/bin/bash
 source ./common/utils.sh
 
-kubectl create namespace istio-system
+kubectl get ns istio-system
+ISTIO_AVAILABLE=$?
+if [[ "$ISTIO_AVAILABLE" == 0 ]] && [[ "$ISTIO_INSTALL_OPTION" == "Reuse" ]]; then
+    # An istio-version is already installed
+    print_info "Istio installation is reused but its compatibility is not checked"
```
Comment (christian-kreuzberger-dtx) [inline installer/scripts/common/setupIstio.sh]: Would it make sense to perform some basic checks here, same as what we do after installation, e.g.:
```bash
 wait_for_crds "adapters.config.istio.io,attributemanifests.config.istio.io,authorizationpolicies.rbac.istio.io,clusterrbacconfigs.rbac.istio.io,destinationrules.networking.istio.io,envoyfilters.networking.istio.io,gateways.networking.istio.io,handlers.config.istio.io,httpapispecbindings.config.istio.io,httpapispecs.config.istio.io,instances.config.istio.io,meshpolicies.authentication.istio.io,policies.authentication.istio.io,quotaspecbindings.config.istio.io,quotaspecs.config.istio.io,rbacconfigs.rbac.istio.io,rules.config.istio.io,serviceentries.networking.istio.io,servicerolebindings.rbac.istio.io,serviceroles.rbac.istio.io,sidecars.networking.istio.io,templates.config.istio.io,virtualservices.networking.istio.io"
```

Please also state in the output that this option is not supported

File: installer/scripts/common/setupIstio.sh [hunk 3]
```diff
@@ -1,23 +1,39 @@
 #!/bin/bash
 source ./common/utils.sh
 
-kubectl create namespace istio-system
+kubectl get ns istio-system
```
Comment (christian-kreuzberger-dtx) [inline installer/scripts/common/setupIstio.sh]: Is istio **always**  installed into `istio-system`, or is there a way to determine if istio is installed into another namespace?

File: installer/scripts/common/setupIstio.sh (hunk 1 above)
Comment (agrimmer) [reply installer/scripts/common/setupIstio.sh]: Thanks, good point! Addressed.

File: installer/scripts/common/setupIstio.sh (hunk 2 above)
Comment (agrimmer) [reply installer/scripts/common/setupIstio.sh]: I do not add this check as we do not know, which CRDs are actually used.

File: installer/scripts/common/setupIstio.sh (hunk 3 above)
Comment (agrimmer) [reply installer/scripts/common/setupIstio.sh]: Istio can be installed in any namespace. However, in this first version, we only check for the `istio-system` namespace.

**Pull request merged**